bin/lib/common.sh

```diff
@@ -0,0 +1,318 @@
+#!/bin/bash
+
+# shellcheck disable=SC2181
+# -- https://www.shellcheck.net/wiki/SC2181
+
+set -eu
+
+declare DEBUG=${DEBUG:-0}
+declare PROJECT_DIR; PROJECT_DIR="$(git rev-parse --show-toplevel 2> /dev/null)" || \
+    (echo_err "$(basename "${0}") commands use git to set directory context. Exiting..." && exit 135)
+if [[ -z "${PROJECT_DIR}" ]]; then
+  echo_err "$(basename "${0}") commands should be run from the directory context of the OSMT git repo. Exiting..."
+  exit 135
+fi
+
+# new line formatted to indent with echo_err / echo_info etc
+declare -r INDENT="       "
+declare -r NL="\n${INDENT}"
+
+source_env_file() {
+  local env_file="${1}"
+
+  if [[ ! -f "${env_file}" ||  ! -r "${env_file}" ]]; then
+    echo_err "Can not access ${env_file}. You can initialize the environment files by running $(basename "${0}") -i$"
+    return 1
+  fi
+
+  _validate_env_file "${env_file}" || return 1
+
+  echo_info "Sourcing ${env_file}"
+  set -o allexport
+  # shellcheck source=/dev/null
+  source "${env_file}"
+  set +o allexport
+}
+
+source_osmt_env_file() {
+  local env_file="${1}"
+
+ # gracefully bypass sourcing env file if these 4 OAUTH values are provided
+  if [[ \
+      -n "${OAUTH_ISSUER}" && \
+      -n "${OAUTH_CLIENTID}" &&\
+      -n "${OAUTH_CLIENTSECRET}" && \
+      -n "${OAUTH_AUDIENCE}" \
+    ]]; then
+      echo_info "OAUTH values are provided by environment variables. Not sourcing ${env_file} env file."
+      return 0
+    fi
+
+  source_env_file "${env_file}"
+}
+
+start_osmt_docker_stack() {
+  local stack_name="${1}"
+  local -i rc
+  echo
+  echo_info "Starting OSMT ${stack_name} Docker stack. You can stop it with $(basename "${0}") -e"
+  cd "${PROJECT_DIR}/docker" || return 1
+  # Docker stack should receive the service port variables sourced from the shell environment
+  docker-compose --file dev-stack.yml --project-name "${stack_name}" up --detach
+  rc=$?
+  if [[ $rc -ne 0 ]]; then
+    echo_err "Starting OSMT ${stack_name} Docker stack failed. Exiting..."
```

Comment:
I think this sentence could be reduced to be just "Docker stack start failed. Exiting.." as it is, seems like we are trying to start the docker stack for a second time since we are already saying in line 58 that the stack is being started..